packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -12,6 +12,25 @@ import 'package:material_color_utilities/material_color_utilities.dart';
 import 'colors.dart';
 import 'theme_data.dart';
 
+/// The algorithm used to construct [ColorScheme] in [ColorScheme.fromSeed].
+///
+/// The `tonalSpot` builds default Material scheme colors. These colors are
+/// mapped to tones (lightness or darkness) to achieve visually accessible color
+/// pairings with sufficient contrast between foreground and background elements.
+///
+/// In some cases, these tones can prevent colors from appearing as intended,
+/// such as when a color is too light to appear vibrant. Color fidelity (`Variant.fidelity`)
+/// is a feature that adjusts tones in these cases to produce the intended visual
+/// results without harming visual contrast.
+enum Variant {
```

Comment:
Although this is a sensible name, it's generic enough to risk colliding with existing code.  Maybe `FromSeedVariant`?

Reply:
Do you think if `DynamicSchemeVariant` would be better? These variants are just used to create different types of `DynamicScheme`, such as `SchemeContent`, `SchemeFidelity`, `SchemeNeutral` etc. They are all subclasses of `DynamicScheme`🙂.

Reply:
DynamicSchemeVariant sounds OK to me; I don't think it will collide with an existing name 👍